scripts/quests/windurst/The_Tenshodo_Showdown.lua

```diff
@@ -0,0 +1,153 @@
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [10021] = function(player, csid, option, npc)
+                    player:delKeyItem(xi.ki.LETTER_FROM_THE_TENSHODO)
+                    npcUtil.giveKeyItem(player, xi.ki.TENSHODO_ENVELOPE)
+                    quest:setVar(player, 'Prog', 1)
+                end,
+
+                [10022] = function(player, csid, option, npc)
+                    if quest:complete(player) then
+                        player:delKeyItem(xi.ki.SIGNED_ENVELOPE)
+                    end
+                end,
+            },
+        },
+
+        [xi.zone.SELBINA] =
+        {
+            ['Elfriede'] =
+            {
+                onTrade = function(player, npc, trade)
+                    if
+                        npcUtil.tradeHasExactly(trade, xi.items.BOWL_OF_QUADAV_STEW) and
+                        quest:getVar(player, 'Prog') == 2
+                    then
+                        return quest:progressEvent(10004, 0, xi.ki.TENSHODO_ENVELOPE, xi.items.BOWL_OF_QUADAV_STEW)
+                    end
+                end,
+
+                onTrigger = function(player, npc)
+                    local questProgress = quest:getVar(player, 'Prog')
+
+                    if questProgress == 1 then
+                        return quest:progressEvent(10002, 0, xi.ki.TENSHODO_ENVELOPE, xi.items.BOWL_OF_QUADAV_STEW)
+                    elseif questProgress == 2 then
+                        return quest:progressEvent(10003, 0, 0, xi.items.BOWL_OF_QUADAV_STEW)
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [10002] = function(player, csid, option, npc)
+                    quest:setVar(player, 'Prog', 2)
+                end,
+
+                [10004] = function(player, csid, option, npc)
+                    player:confirmTrade()
+                    player:delKeyItem(xi.ki.TENSHODO_ENVELOPE)
+                    npcUtil.giveKeyItem(player, xi.ki.SIGNED_ENVELOPE)
+                    quest:setVar(player, 'Prog', 3)
+                end,
+            },
+        },
+
+        [xi.zone.WINDURST_WOODS] =
+        {
+            ['Nanaa_Mihgo'] =
+            {
+                onTrigger = function(player, npc)
+                    local questProgress = quest:getVar(player, 'Prog')
+
+                    if questProgress == 0 then
+                        return quest:progressEvent(497)
+                    elseif questProgress == 1 then
+                        return quest:progressEvent(498)
+                    end
+                end,
+            },
+        },
+    },
+
+    {
+        check = function(player, status, vars)
+            return status == QUEST_COMPLETED and
+                player:getMainJob() == xi.job.THF and
+                player:getMainLvl() < xi.settings.AF2_QUEST_LEVEL
+        end,
```

Comment:
Could you add a check for the next quest here (as quest available).
Unlikely BUT, what if a player were to repeatedly delevel? they would hit this after completing the next quest.